contracts/core/EntryPoint.sol

```diff
@@ -113,47 +116,68 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
                 innerCall = abi.encodeCall(this.innerHandleOp, (callData, opInfo, context));
             }
             assembly ("memory-safe") {
-                success := call(gas(), address(), 0, add(innerCall, 0x20), mload(innerCall), 0, 32)
-                collected := mload(0)
+                success := call(gas(), address(), 0, add(innerCall, 0x20), mload(innerCall), 0, 64)
+                returnData := mload(0)
+                actualGas := mload(32)
             }
             restoreFreePtr(saveFreePtr);
         }
-        if (!success) {
-            bytes32 innerRevertCode;
-            assembly ("memory-safe") {
-                let len := returndatasize()
-                if eq(32,len) {
-                    returndatacopy(0, 0, 32)
-                    innerRevertCode := mload(0)
-                }
-            }
+        bool callSuccess;
+        if (success) {
+            callSuccess = returnData != 0;
+        } else {
+            bytes32 innerRevertCode = returnData;
             if (innerRevertCode == INNER_OUT_OF_GAS) {
                 // handleOps was called with gas limit too low. abort entire bundle.
                 // can only be caused by bundler (leaving not enough gas for inner call)
                 revert FailedOp(opIndex, "AA95 out of gas");
             } else if (innerRevertCode == INNER_REVERT_LOW_PREFUND) {
-                // innerCall reverted on prefund too low. treat entire prefund as "gas cost"
-                uint256 actualGas = preGas - gasleft() + opInfo.preOpGas;
-                uint256 actualGasCost = opInfo.prefund;
-                emitPrefundTooLow(opInfo);
-                emitUserOperationEvent(opInfo, false, actualGasCost, actualGas);
-                collected = actualGasCost;
+                // innerCall reverted on prefund too low.
+                // below we re-calculate the cast cost and refund.
```

Comment:
Where? Put function name, lines or some other meaningful reference to what you mean here.

Also, this entire `if` is empty, could we maybe "invert" it to be `else if (!innerRevertCode == INNER_REVERT_LOW_PREFUND) { ... } ` ??